### ISSUE TITLE
Into Bluesky: images should have aspectRatio, otherwise preview is square

### ISSUE THREAD

**Issue by MS-potilas** [snarfed/bridgy-fed#1571]
Related to https://github.com/bluesky-social/social-app/pull/6474

If aspectRatio (width, height) is not set, images are previewed as clipped square. Examples here https://bsky.app/profile/fok-it.eliitin-some.fi.ap.brid.gy

Bluesky client app sets the aspectRatio, but most other clients (including blue deck) and API users don't (and now they're supposed to).

**Comment by snarfed** [comment]
Good point! Specifically, this is the `aspectRatio` field in `app.bsky.embed.images`.

Sadly, this is somewhat unlikely. I don't currently do any image processing at all, and I'm reluctant to start. I can keep this open to track though!

**Comment by ulrischa** [comment]
Same issue here https://bsky.app/profile/lfu.bayern.de

**Comment by snarfed** [comment]
@ulrischa looks like that's a normal Bluesky account, not a bridged account.

**Comment by ulrischa** [comment]
@snarfed yes it is. The problem has nothing to do with bridged or not. There seems to be a breaking change in aspect ratio handling

**Comment by roddie-digital** [comment]
For info, Bluesky have implemented a (hopefully temporary) workaround for this in [social-app#6828](https://github.com/bluesky-social/social-app/pull/6828) - images are now displayed in their correct aspect ratio but within a square for their preview. eg:

https://bsky.app/profile/ghostintheshell.roddie.social/post/3lcjthp4mota2

**Comment by popet-master** [comment]
So eventually this issue, unrelated to bridgy, gets fixed by Bluesky, as a plain `css` bug on their side, right ?

**Comment by snarfed** [comment]
No, I don't think so. They now letterbox instead of cropping so that they don't cut anything out, but they still expect clients to all eventually set `aspectRatio`.

This isn't a high priority for Bridgy Fed right now, but PRs are welcome, feel free to try!

**Comment by Tamschi** [comment]
I think it's reasonable for them to do it this way with their architecture, as it decreases service dependencies significantly for them.

**Comment by snarfed** [comment]
@JoelOtter is currently adding this to granary in https://github.com/snarfed/granary/pull/844, that will do half of this and give us an easy template for the other half. Thank you Joel!

**Comment by MS-potilas** [comment]
Can the Granary addition be deployed "as is" to Bridgy Fed, too?

**Comment by snarfed** [comment]
As described in https://github.com/snarfed/bridgy-fed/issues/1571#issuecomment-2536851194, it's half of the solution! The other half is porting https://github.com/snarfed/granary/pull/844/files#diff-1915ad0740506cba5b100e33d222098b9ec534f9f592b600ab55571db03296b0L2175-R2288 to `ATProto._convert` in `atproto.py` here.

**Comment by snarfed** [comment]
Done! Example: https://mastodon.social/@martinicat/113790819522277326 => https://bsky.app/profile/martinicat.mastodon.social.ap.brid.gy/post/3lf7cudvaxzx2
